### PR TITLE
Add an exception handler for the pubsub thread

### PR DESCRIPTION
**Pull request by fcharlier**
Hi,

This is proposal to fix issues #1836 & #2070

After a connection error to Redis the pubsub thread was stopped and the worker could not receive commands anymore.

This commit adds an exception handler for the thread that adds a log message and ignores `redis.exceptions.ConnectionError`. Any other exception is re-raised.

redis-py internal mechanism allows the pubsub thread to recover its connection and reinstall the pubsub channel subscription to allow the worker to receive commands again after connection errors.

It tries to behave the same as the main worker loop retry mechanism but without the backoff wait factor because we're not operating in a loop.

### CODE REVIEW

**Comment by fcharlier** [comment]
N.B.: I have tested this change on rq==1.16.2 but this part of the code seems almost untouched in master.

File: rq/worker.py [hunk 1]
```diff
@@ -950,12 +950,33 @@ def run_maintenance_tasks(self):
...
+        """
+        if isinstance(exc, (redis.exceptions.ConnectionError)):
+            self.log.error(
+                "Could not connect to Redis instance: %s Retrying in %d seconds...",
+                exc,
+                2,
+            )
+            print("Connection Error, waiting 2 seconds before retry")
+            time.sleep(2.0)
+        else:
```
Comment (selwin) [inline rq/worker.py:969]: For completeness' sake, is there an easy way to cover this line in test?

Comment (fcharlier) [reply rq/worker.py:968]: @selwin new commit should be fine.

**Comment by selwin** [comment]
Thanks!

**Comment by selwin** [comment]
@fcharlier will release 2.0 end of this month so stay tuned :)

**Pull request merged**